niftynet/network/dense_vnet.py

```diff
@@ -415,6 +415,23 @@ def layer_op(self, input_tensor, is_training=None, keep_prob=None):
 
             stack.append(conv)
 
+        # unmask the stack
+        channels = tf.unstack(stack, axis=-1)
+        # exclude the input channels
+        channels = channels[input_channels:]
```

Comment:
Thanks for the fix! 
This `input_channels` is undefined, perhaps `input_channels = input_tensor.shape.as_list()[-1]`?

Reply:
Yes, sorry, that is correct `input_channels = input_tensor.shape.as_list()[-1]`. This can be defined earlier on and used here as well as in the one other part of the code that requires it.

Reply:
Hi @MiguelMonteiro I don't quite understand the changes here,  e.g. `stack` is a list of tensors `tf.unstack(stack, axis=-1)` doesn't make sense to me.  Could you please give me a working example, or make it work for the model zoo entry?
```
python net_download.py dense_vnet_abdominal_ct_model_zoo
python net_segment inference -c ~/niftynet/extensions/dense_vnet_abdominal_ct/config.ini
```

Reply:
ups, the toy program I was developing already used `tf.concat` instead of appending to a list, hence the line `tf.unstack(stack, axis=-1)`.  If you remove that line the rest should work fine because the stack is already a list.

Reply:
Here is my toy program just as reference (sorry for the variables having different names):

```
import tensorflow as tf
# the stack must be unmasked here
stack = tf.ones(shape=(5, 40, 40, 40, 10))
input_mask = tf.constant([True, True, False, False, True, True, True, True, True, True, True, True, False])
input_channels = 2
masked_channels = tf.unstack(stack, axis=-1)
masked_channels = masked_channels[input_channels:]  # exclude the input
masked_channels.insert(0, tf.zeros_like(masked_channels[0]))
masked_channels = tf.stack(masked_channels, axis=-1)

input_mask = input_mask[input_channels:]  # exclude the input
# indices to keep, shift by one to include the zeroed out channel in the first position of the list
indices = tf.cumsum(tf.cast(input_mask, tf.int32)) * tf.cast(input_mask, tf.int32)
unmasked_stack = tf.gather(masked_channels, indices, axis=-1)
```